lib/improver/cli/probabilities_to_realizations.py

```diff
@@ -37,101 +37,95 @@
 @cli.clizefy
 @cli.with_output
 def process(cube: cli.inputcube,
-            raw_forecast: cli.inputcube = None,
+            raw_cube: cli.inputcube = None,
             *,
-            no_of_realizations: int = None,
-            reordering=False,
-            rebadging=False,
+            realizations_count: int = None,
+            reorder=False,
+            rebadge=False,
             random_seed: int = None,
-            ecc_bounds_warning=False):
+            ignore_ecc_bounds=False):
     """Convert from probabilities to ensemble realizations.
 
     Args:
         cube (iris.cube.Cube):
             Cube to be processed.
-        raw_forecast (iris.cube.Cube):
+        raw_cube (iris.cube.Cube):
             Cube of raw (not post processed) weather data.
-            This option is compulsory, if the reordering option is selected.
-        no_of_realizations (int):
+            This option is compulsory, if the reorder option is selected.
+        realizations_count (int):
             Optional definition of the number of ensemble realizations to
             be generated. These are generated though an intermediate
             percentile representation. Theses percentiles will be
             distributed regularly with the aim of dividing into blocks
-            of equal probability. If the reordering option is specified
+            of equal probability. If the reorder option is specified
             and the number of realization is not given the number
             of realizations is taken from the number of realizations
             in the raw forecast cube.
-            Default is None.
-        reordering (bool):
+        reorder (bool):
             The option used to create ensemble realizations from percentiles
             by reordering the input percentiles based on the order of the
             raw ensemble.
-            Default is False.
-        rebadging (bool):
+        rebadge (bool):
             Th option used to create ensemble realizations from percentiles
             by rebadging the input percentiles.
```

Comment:
This looks like a simple two-boolean option CLI to me.  Which means we only need one of these options; one must be true and one must be false, otherwise `result` is not defined and the CLI would throw an error on `return`.  Or am I missing something here?

Reply:
Yes, I think it's correct that `reorder` and `rebadge` could be condensed into a single CLI argument, if preferred.

Reply:
Which one should be the default? Is this also true for `percentiles_to_realizations`?

Reply:
After discussing this with @gavinevans, it turns out we can drop both of these options (in both CLIs) and use the presence (absence) of `raw_cube` argument to reorder (rebadge) the result.

Looking into this a bit more, I think that out of `convert_to_realizations`, `percentile`, `percentiles_to_realizations` and `probabilities_to_realizations` we only really need two, e.g.: `percentile` and `percentiles_to_realizations`, because the other two are effectively build out of them.

I don't have the time to go through this right now, but we could look into this as part of `ver_stream_preprocessing` work, so I'm tagging @benfitzpatrick here.

Reply:
Yes, technically, but I'm reluctant to make a two step process out of a one step - I actually prefer utilities like convert-to-realizations that have a clear end goal - I think we should have a convert-to-probabilities and a convert-to-percentiles as well and drop all the other CDF ones.

Another argument is that in the case of percentile and percentiles_to_realizations, there is an unnecessary computational cost in doing a CDF conversion twice.

Reply:
Nice.  These are much tidier now and easier to understand.